File: lib/server/file/deleteAllFileFromDirectory.ts
import { readdir } from 'node:fs/promises';
import { join } from 'path';
import { deleteFile } from './deleteFile.js';

export async function deleteAllFileFromDirectory(directoryPath: string): Promise<void> {
	let fileItems: Array<string> = [];

	try {
		fileItems = await readdir(directoryPath);
	} catch {
		// Do nothing
	}

	const fileItemLength: number = fileItems.length;

	for (let i = 0; i < fileItemLength; i += 1) {
		// eslint-disable-next-line no-await-in-loop
		await deleteFile(join(directoryPath, fileItems[i]));
	}
}

File: packages/python/qsu/file/deleteAllFileFromDirectory.py
import os
from .deleteFile import deleteFile


def deleteAllFileFromDirectory(directoryPath: str) -> None:
	fileItems = []

	try:
		fileItems = os.listdir(directoryPath)
	except Exception:
		# Do nothing
		pass

	for item in fileItems:
		deleteFile(os.path.join(directoryPath, item))
